lib/provider/api.ts

```diff
@@ -0,0 +1,84 @@
+import {encode} from 'cbor-x'
+import {IARC0001Transaction} from "./types";
+import {RequestMessageBuilder} from "./models/request.message.js";
+import {SignTransactionsParamsBuilder} from "./models/sign.transactions.params.js";
+import {SignTransactionsResultBuilder} from "./models/sign.transactions.result.js";
+import {ResponseMessageBuilder} from "./models/response.message.js";
+
+const LARGE_MESSAGE_SIZE = 255_000;
+const LARGE_MESSAGE_ERROR = `Message too large, maximum is ${LARGE_MESSAGE_SIZE} in length`
+
+const chars =
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_';
+export const INVALID_BASE64URL_INPUT = 'Invalid base64url input';
+
+
+/**
+ * Bytes to Base64URL
+ * @param {Uint8Array| ArrayBuffer} arr Bytes to convert to URL safe Base64
+ */
+export function toBase64URL(arr: Uint8Array | ArrayBuffer): string {
```

Comment:
Isn't there something that already does this?

Reply:
Sadly there is not a cross platform native way for Base64 URL. `Buffer` only exists in Node.js and not the Browser.

Uint8Array/ArrayBuffer are available in both contexts but lack any string serializers like Buffer. The existing packages I could find where out of date and small enough in scope to just implement our own version.

Reply:
cool makes sense